### DISCUSSION TITLE
How to run Adonis without a .env file in kubernetes

### DISCUSSION THREAD

**Discussion by simonjcarr** [adonisjs/core#2300] [Help] [answered]
I need to run Adonis in Kubernetes. My docker images are built via a GitLab CI/CD pipeline. The only code available to the pipeline is that which is in the repository, and we know that we should not push .env files into git that contain passwords.

I have all the ENV variables I need set in my Dockerfile, but Kubernets can not build the image without the .env file.

```
Error: ENOENT: no such file or directory, open '/. /app/.env'
```

What is the correct process for using Adonis with CI/CD and Kubernetes?

**Comment by RomainLanz** [comment, marked as answer]
Hey @simonjcarr! 👋 

You need to provide the `ENV_SILENT=true` environment variable.

```bash
ENV_SILENT=true node server.js
```

(Same applies for V4 or V5)

📚  https://adonisjs.com/docs/4.1/configuration-and-env#_disabling_the_env_file

**Reply by simonjcarr** [reply]
@RomainLanz Thanks for the quick response.

So the following should work for me?

```
ENV ENV_SILENT=true
ENV HOST=0.0.0.0
ENV PORT=3333
ENV NODE_ENV=production
ENV APP_NAME=blog-api
ENV APP_URL=http://localhost:3333
ENV CACHE_VIEWS=false
ENV APP_KEY=xxxxxxxxxxxxxx
ENV DB_CONNECTION=pg
ENV DB_HOST=postgres
ENV DB_PORT=5432
ENV DB_USER=blog
ENV DB_PASSWORD=xxxxxxxx
ENV DB_DATABASE=blog
ENV HASH_DRIVER=bcrypt

CMD npm run start

```

**Reply by simonjcarr** [reply]
The following worked as described by @RomainLanz in my Dockerfile

```
CMD ENV_SILENT=true node server.js
```
